content/en/docs/Installation/trustedboot.md

```diff
@@ -121,6 +121,44 @@ title My Awesome OS
 Remember when you build upgrade mediums, to also add your branding, otherwise the new active image will be named differently than your fallback and recovery ones.
 {{% /alert %}}
 
+### Additional efi entries
+
+When booting in UKI mode, the cmdline is part of the signed artifact. This means that in order to pass additional options through the cmdline,
+an additional efi entry is needed. `enki build-uki` supports the following arguments:
+
+- `--extend-cmdline`: Use this option to add options to the cmdline used for active/passive/recovery. This option will not generate additional entries in the systemd-boot menu.
+- `--extra-cmdline`: Use this option to generate additional entries in the systemd-boot menu. For each for active/passive/recovery, a new entry will be created with the specified cmdline (added to the default cmdline).
+- `--single-efi-cmdline`: Use this option to generate entries that are not related to active/passive/recovery. This option will generate a single entry with the provided cmdline (added to the default cmdline). This argument can be used multiple times.
+
+
+Examples:
+
+Building with a command like this:
+
+```
+enki build-uki ...more options... --single-efi-cmdline "Lets debug: rd.debug rd.immucore.debug"
```

Comment:
TBH I'd really like to push back a bit with this way of setting the title of the single entry. I find it very confusing that for ALL other entries we use a single `--boot-branding` and now for this one, we have this special setup. So what happens when I do:

```
--boot-branding="AwesomeOS" --extend-cmdline="foo" --single-efi-cmdline="Debug: rd.debug"
```

Then I will have:

- AwesomeOS
- AwesomeOS (fallback)
- AwesomeOS recovery
- AwesomeOS reset (auto)
- Debug

Reply:
@jimmykarily I think we should go back to product or at least the edge team, they requested this feature so they should be the one telling us if the branding is also expected on the single efi entry, but IMO branding should impact all entries including the single ones.

I don't think having to match flags is going to be tricky in the code, but maybe I'm missing something? Otherwise, then maybe it should actually be named `--single-efi-config=./path/to/my-efi.config` where `my-efi.config` includes: title, cmdline, more?

Reply:
@vipsharm any thoughts?

Reply:
Vipin said they'd like the branding to also apply to the "single" entries. This doesn't answer the question of how we set the unique part of the title though. So the user still needs to specify some text to be appended to the title. E.g. With `boot-branding` set to "My OS", the user needs a way to produce entries like:

```
My OS (debugging)
My OS (reset)
```

With a small modification in the current code, we can achieve this with:

```
enki build-uki ... --boot-branding "My OS" --single-efi-cmdline "debugging: rd.debug" --single-efi-cmdline "reset: someresetoption"
```

If we were to split the title and cmdline parts, we would need something like:

```
enki build-uki ... --boot-branding "My OS" --single-efi-cmdline "rd.debug" --single-efi-cmdline "someresetoption" --single-efi-title "debugging" --single-efi-title "reset"
```

and then we match the first cmdline with the first title etc. What happens when the user provides less/more titles than cmdlines? We can show a validation error, like we do when the title part is missing in the current implementation.

@mauromorales do you think this is better UX? I don't have a strong preference.

Reply:
@jimmykarily if the branding applies to everything, I'm also ok with the original approach of `--single-efi-cmdline "debugging: rd.debug"` because I think it will be less likely that someone will want to put a more complex name there. So let's go with your first option. One last question, what will happen if I don't pass a name? Keep in mind that systemd-boot has its own way of differientiating boot entries that are named the same

Reply:
ok let me try things

Reply:
This: 

```
--boot-branding "KairosUKI" --single-efi-cmdline "My entry: someoption=somevalue" --single-efi-cmdline "Other entry: otheroption=othervalue" 
```

now produces:
<img width="255" alt="image" src="https://github.com/kairos-io/kairos-docs/assets/2794419/5a8d254c-8ecc-4a95-a578-af493f1fbc3e">

I don't know if we want to do something about the ordering.

Reply:
If we were to allow empty titles for the single entries, this would happen:

<img width="302" alt="image" src="https://github.com/kairos-io/kairos-docs/assets/2794419/ce8a4448-fe0c-4413-8d40-b29eaee7ed62">

(I hardcoded the title to "samething" to trigger this). Let's allow this then.

Reply:
yeah I think the only problem with allowing that is that the string becomes so long (because of the long branding name used by product) that they cannot even read the last parts, but they might as well just name it something different :P